Support for trackSiteSearch

MikiDi opened the pull request:
Hi there,

We make use of the Matomo method [`trackSiteSearch`](https://developer.matomo.org/guides/tracking-javascript-guide#internal-search-tracking), which doesn't map to any of `ember-tracking`'s [main api methods](https://github.com/adopted-ember-addons/ember-metrics#api). This PR provides support for this Matomo-specific method. It now can be used through `ember-tracking`'s `invoke`. See the Readme changes for more information. :-)

Regards,

Michaël

MikiDi committed:
feat: add matomo-specific trackSiteSearch method

LevelbossMike commented:
This looks great! Thanks for putting this together. Can you please add a test for this behavior and restructure the commits so that the `paq` change goes together with the `trackSiteSearch`-implementation. That way we have a coherent commit history.

Thanks again for putting the effort in to add this feature :beers:

MikiDi commented:
@LevelbossMike You [mentioned](https://discord.com/channels/480462759797063690/484421406659182603/766366577448452178) something on discord about types not being complete. I must admit that this is my very first attempt at adapting some typescript code :innocent: ... Could you elaborate a bit please :pray:   
I ran `npm run prepack` and `npm run postpack` in the repo itself and that didn't throw any warnings ...

LevelbossMike commented:
@MikiDi Ah sorry I just saw the linting errors on https://travis-ci.org/github/effective-ember/ember-metrics-matomo-adapter/builds/736037082 and figured it also included typescript errors.

Should be fine then - but we should fix the linting errors so that we see that tests are running through. Other than that this looks good to me

MikiDi commented:
Ok, thanks, will fix those. (Last time I looked, ci grinded to halt on installing yarn, looked like rate limiting on the download ... )

MikiDi commented:
Fixed the linting errors, squashed and reworded. Didn't touch the existing warnings about the `any` typings though.
_edit: Forgot the test though ... will take care of that_

MikiDi commented:
Added a test. Should be good now :sweat_smile:

LevelbossMike reviewed (approved):
Thanks for putting this together :+1:

LevelbossMike commented:
@MikiDi I'm busy today so I will release tomorrow - please ping me if I forget to do so

LevelbossMike commented:
@MikiDi released as 0.2.0 - thanks again for putting in the effort for landing this feature.
